BraveWallet/Crypto/BuySwapSwap/SwapCryptoView.swift

```diff
@@ -40,38 +69,151 @@ struct ShortcutAmountGrid: View {
+            if selectedIndex != option.index && changed {
+              selectedIndex = option.index
+            }
+          })
+            .onChange(of: input) { value in
+              if selectedIndex == option.index {
+                action((Double(value) ?? option.value) / 100)
+              }
+            }
+            .keyboardType(.numberPad)
+            .multilineTextAlignment(.center)
+            .lineLimit(1)
+            .minimumScaleFactor(0.75)
+            .padding(.vertical, 12)
+            .frame(maxWidth: .infinity)
+            .foregroundColor(Color(option.index == selectedIndex ? .white : .secondaryBraveLabel))
+            .modifier(RroundedBackgroundModifier(backgroundColor: Color(option.index == selectedIndex ? .braveLighterBlurple : .secondaryBraveGroupedBackground)))
+            .padding(.top, 8)
+        } else {
+          Button(action: {
+            if selectedIndex != option.index {
+              selectedIndex = option.index
+            }
+            input = ""
+            action(option.value)
+            resignFirstResponder()
+          }) {
+            Text(option.label)
+              .lineLimit(1)
+              .minimumScaleFactor(0.75)
+              .padding(.vertical, 12)
+              .frame(maxWidth: .infinity)
+              .foregroundColor(Color(option.index == selectedIndex ? .white : .secondaryBraveLabel))
+              .modifier(RroundedBackgroundModifier(backgroundColor: Color(option.index == selectedIndex ? .braveLighterBlurple : .secondaryBraveGroupedBackground)))
+              .padding(.top, 8)
+          }
+        }
+      }
+    }
+  }
+  
+  func resignFirstResponder() {
+      UIApplication.shared.sendAction(#selector(UIResponder.resignFirstResponder), to: nil, from: nil, for: nil)
+  }
+}
+
+struct MarketPriceView: View {
+  @ObservedObject var swapTokenStore: SwapTokenStore
+  
+  var body: some View {
+    HStack {
+      VStack(alignment: .leading) {
+        Text(String.localizedStringWithFormat(Strings.Wallet.swapCryptoMarketPriceTitle, swapTokenStore.selectedFromToken?.symbol ?? ""))
+          .foregroundColor(Color(.secondaryBraveLabel))
+          .font(.subheadline)
+        Text("$\(swapTokenStore.selectedFromTokenPrice ?? 0)")
+          .font(.title3.weight(.semibold))
+      }
+      Spacer()
+      Button(action: { }) {
+        Image("wallet-refresh")
+          .renderingMode(.template)
+          .foregroundColor(Color(.braveLighterBlurple))
+          .font(.title3)
+      }
+      .buttonStyle(.plain)
+    }
+  }
+}
+
 struct SwapCryptoView: View {
   @ObservedObject var keyringStore: KeyringStore
   @ObservedObject var ethNetworkStore: NetworkStore
+  @ObservedObject var swapTokensStore: SwapTokenStore
   
   @State private var fromQuantity: String = ""
   @State private var toQuantity: String = ""
   @State private var orderType: OrderType = .market
+  @State var slippage: Double = 0.005
```

Comment:
default slippage tolerance as 0.5%

Reply:
As mentioned use the actual Option type not a value